src/main/java/de/tum/in/www1/artemis/repository/tutorialgroups/TutorialGroupRegistrationRepository.java

```diff
@@ -0,0 +1,22 @@
+package de.tum.in.www1.artemis.repository.tutorialgroups;
+
+import java.util.Optional;
+import java.util.Set;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.domain.tutorialgroups.TutorialGroup;
+import de.tum.in.www1.artemis.domain.tutorialgroups.TutorialGroupRegistration;
+
+@Repository
+public interface TutorialGroupRegistrationRepository extends JpaRepository<TutorialGroupRegistration, Long> {
+
+    Optional<TutorialGroupRegistration> findTutorialGroupRegistrationByTutorialGroupAndStudent(TutorialGroup tutorialGroup, User student);
```

Comment:
Regarding my comment in the migration file:
If you allow multiple entries for a single (student, tutorialGroup) pair, then this query will not work reliably. If there is more than one entry for such a pair it will throw this exception:
`javax.persistence.NonUniqueResultException: query did not return a unique result: 2`.

Considering that this PR does not include the self-registration functionality, this should not be an issue currently, but once you implement it and allow multiple entries to exist, then it will become a problem.

Considering this you don't have to fix this issue in this PR, but considering that it might require bigger changes you should consider it.

Reply:
Good catch! I noted this down and will consider this once we implement self registration. Thank you 👍